src/goto-instrument/contracts/utils.cpp

```diff
@@ -376,7 +376,9 @@ void add_quantified_variable(
     auto &unary_expression = to_unary_expr(expression);
     add_quantified_variable(symbol_table, unary_expression.op(), mode);
   }
-  if(expression.id() == ID_notequal || expression.id() == ID_implies)
+  if(
```

Comment:
Would it be possible to transform `forall` and `exists` nodes without having to explicitly match on other `irep_ids` ? Maybe using `exprt::visit_post` or `exprt::visit_pre` ?

Reply:
Rewrote the whole function with `visit_pre`.